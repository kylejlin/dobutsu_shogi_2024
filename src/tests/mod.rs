use super::*;

mod i9;

mod helpers;

use helpers::IntoPretty;

#[test]
fn initial_search_node_is_correct() {
    insta::assert_snapshot!(SearchNode::initial().pretty());
}

#[test]
fn initial_search_node_allegiance_inversion_is_correct() {
    insta::assert_snapshot!(SearchNode::initial()
        .into_builder()
        .invert_active_player()
        .pretty());
}

#[test]
fn initial_search_node_with_incremented_ply_count_is_correct() {
    insta::assert_snapshot!(SearchNode::initial()
        .into_builder()
        .increment_ply_count()
        .pretty());
}

#[test]
fn initial_search_node_with_initialized_best_discovered_outcome_and_next_action() {
    insta::assert_snapshot!(SearchNode::initial()
        .into_builder()
        .init_best_discovered_outcome_and_next_action()
        .pretty());
}

#[test]
fn initial_search_node_horizontally_flipped_is_correct() {
    insta::assert_snapshot!(SearchNode::initial()
        .into_builder()
        .horizontally_flip()
        .pretty());
}

#[test]
fn initial_search_node_allegiance_inversion_built_is_correct() {
    insta::assert_snapshot!(SearchNode::initial()
        .into_builder()
        .invert_active_player()
        .build()
        .pretty());
}

#[test]
fn initial_search_node_children_are_correct() {
    insta::assert_snapshot!(SearchNode::initial().children().pretty());
}
