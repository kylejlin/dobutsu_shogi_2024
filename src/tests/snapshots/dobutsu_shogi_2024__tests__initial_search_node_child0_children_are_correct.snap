---
source: src/tests/mod.rs
expression: "format!(\"parent:\\n{child0}\\n\\nchildren:\\n{children}\")"
---
parent:
|===|                |===|
|...|                |...|
|...|                |...|
|===|                |===|
|---|                |---|
|G*E|                |GLE|
|*CL|                |*C*|
|*c*|                |lc*|
|elg|                |e*g|
|---|                |---|
required_child_report_count: 0
best_known_outcome: 0 (draw)

children:
SearchNodeSet(len = 4) [
    0:
    |===|                |===|
    |...|                |...|
    |...|                |...|
    |===|                |===|
    |---|                |---|
    |G*E|                |G*E|
    |*CL|                |*CL|
    |lc*|                |lc*|
    |e*g|                |e*g|
    |---|                |---|
    required_child_report_count: 0
    best_known_outcome: 0 (draw)
    ==========================
    1:
    |===|                |===|
    |...|                |...|
    |...|                |...|
    |===|                |===|
    |---|                |---|
    |G*E|                |G*E|
    |LC*|                |*CL|
    |lc*|                |*cl|
    |e*g|                |e*g|
    |---|                |---|
    required_child_report_count: 0
    best_known_outcome: 0 (draw)
    ==========================
    2:
    |===|                |===|
    |C..|                |c..|
    |...|                |...|
    |===|                |===|
    |---|                |---|
    |GLE|                |G*E|
    |***|                |*cL|
    |lC*|                |***|
    |e*g|                |elg|
    |---|                |---|
    required_child_report_count: 0
    best_known_outcome: 0 (draw)
    ==========================
    3:
    |===|                |===|
    |...|                |...|
    |...|                |...|
    |===|                |===|
    |---|                |---|
    |*LE|                |G*E|
    |GC*|                |*CL|
    |lc*|                |*cg|
    |e*g|                |el*|
    |---|                |---|
    required_child_report_count: 0
    best_known_outcome: 0 (draw)
    ==========================
]
